main.js

```diff
@@ -1,88 +1,217 @@
+    }
+    childrenInput.value = `${childrenVal} Children`;
+    lableValue.innerHTML  = `${adultsVal} Adults — ${childrenVal} Children — ${roomsVal} Room`;
+}
+
+document.getElementById("counter-children__minus").onclick = function() {
+    childrenColorPlus.style.color = "#3077C6";
+    childrenColorPlus.style.borderColor = "#3077C6";
+    if (childrenVal > 0){
+        childrenValue.value = --childrenVal;
+        let removeChildSelector = selectAge.lastChild;
+        selectAge.removeChild(removeChildSelector);
+    }
+    if(childrenVal === 0){
+        selectAge.style.display = "none";
+        childrenColorMinus.style.color = "#CECECE";
+        childrenColorMinus.style.borderColor = "#CECECE";
+    }
+    childrenInput.value = `${childrenVal} Children`;
+    lableValue.innerHTML  = `${adultsVal} Adults — ${childrenVal} Children — ${roomsVal} Room`;
+}
+
+let roomsValue = document.getElementById("rooms_value");
+let roomsVal = parseFloat(roomsValue.value);
+let roomsColorMinus = document.getElementById("counter-rooms__minus");
+let roomsColorPlus = document.getElementById("counter-rooms__plus");
+let roomsInput = document.getElementById("filter-input__rooms");
+const maxRooms = 30;
+document.getElementById("counter-rooms__plus").onclick = function() {
+    if (roomsVal < maxRooms) {
+        roomsValue.value = ++roomsVal;
+        roomsColorMinus.style.color = "#3077C6";
+        roomsColorMinus.style.borderColor = "#3077C6";
+    }
+    if (roomsVal === maxRooms){
+        roomsColorPlus.style.color = "#CECECE";
+        roomsColorPlus.style.borderColor = "#CECECE";
+    }
+    roomsInput.value = `${roomsVal} Rooms`;
+    lableValue.innerHTML  = `${adultsVal} Adults — ${childrenVal} Children — ${roomsVal} Room`;
+}
+document.getElementById("counter-rooms__minus").onclick = function() {
+    roomsColorPlus.style.color = "#3077C6";
+    roomsColorPlus.style.borderColor = "#3077C6";
+    if (roomsVal > 0){
+        roomsValue.value = --roomsVal;
+    }
+    if(roomsVal === 0){
+        roomsColorMinus.style.color = "#CECECE";
+        roomsColorMinus.style.borderColor = "#CECECE";
+    }
+    roomsInput.value = `${roomsVal} Rooms`;
+    lableValue.innerHTML  = `${adultsVal} Adults — ${childrenVal} Children — ${roomsVal} Room`;
+}
+
+//                  LESSON-12
+let url = 'https://fe-student-api.herokuapp.com/api/hotels/popular'
+fetch(url)
+    .then((response) => {
+        return response.json();
+    })
+    .then((data) => {
+        const dataNew = data.slice(0, 4)
+        let hotelOffer = ''
+        let favoriteHotels = dataNew.reduce((prev, hotel) => {
+            return hotelOffer += `<div class="hotel-foto foto-on col-xl-3 col-sm-6">
                 <div class="hotel-img col-sm-12">
                    <img src="${hotel.imageUrl}" alt="picture">
                 </div>
                 <p>${hotel.name}</p>
                 <p><span>${hotel.city}, ${hotel.country}</span></p>
             </div>`
```

Comment:
Look, you have `prev` as an empty string and you have `hotelOffer` as an empty string. I think you can use one :)